src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationSource.cs

```diff
@@ -29,35 +33,33 @@ public IConfigurationProvider Build(IConfigurationBuilder builder)
             try
             {
                 AzureAppConfigurationOptions options = _optionsProvider();
-                IConfigurationClientManager clientManager;
 
                 if (options.ClientManager != null)
                 {
-                    clientManager = options.ClientManager;
+                    return new AzureAppConfigurationProvider(options.ClientManager, options, _optional);
                 }
-                else if (options.ConnectionStrings != null)
+
+                IEnumerable<Uri> endpoints;
+                IAzureClientFactory<ConfigurationClient> clientFactory = options.ClientFactory;
+
+                if (options.ConnectionStrings != null && options.ConnectionStrings.Any())
                 {
-                    clientManager = new ConfigurationClientManager(
-                        options.ConnectionStrings,
-                        options.ClientOptions,
-                        options.ReplicaDiscoveryEnabled,
-                        options.LoadBalancingEnabled);
+                    endpoints = options.ConnectionStrings.Select(cs => new Uri(ConnectionStringUtils.Parse(cs, ConnectionStringUtils.EndpointSection)));
+
+                    clientFactory ??= new AzureAppConfigurationClientFactory(options.ConnectionStrings.First(), options.ClientOptions);
```

Comment:
I'm still thinking about how we want to handle this implementation of `AzureAppConfigurationClientFactory`. We can't pass the first connection string like this since we still support both APIs that allow passing `IEnumerable` of connection strings or endpoints, and we're not making it clear anywhere that the other connection strings are being ignored. I think this is especially an issue if someone disables replica discovery.

Reply:
Of course, one option is to just add both `Connect` patterns that use `IEnumerable` to this factory.

Reply:
Agree that's a flaw, we will pass all the connection strings

Reply:
I just pushed a new change, the factory stores all the connection strings provided by the customer.

When trying to create a client from an endpoint given provided connection string, we try to find the one that matches the endpoint. Otherwise, we default to the first connection string (replica auto discovery scenario).